proposals/collection-expressions-inline.md

```diff
@@ -0,0 +1,124 @@
+# Collection expressions: inline collections
+
+## Summary
+
+Support collection expressions *inline* in expression contexts where the *collection type* is not observable.
+
+## Motivation
+
+Inline collection expressions could be used with spreads to allow adding elements *conditionally* to the containing collection:
+```csharp
+int[] items = [x, y, .. b ? [z] : []];
+```
+
+Inline collection expressions could be used directly in `foreach`:
+```csharp
+foreach (var b in [false, true]) { }
+```
+In these cases, the *collection type* of the inline collection expression is unspecified. The choice of how or whether to instantiate the collection is left to the compiler.
+
+## Detailed design
+
+### Conversions
+
+To support conversions for inline collection expressions, an abstract *collection_type* is introduced.
+
+A *collection_type* represents an enumerable type with a specific element type and an *unspecified* collection type.
+A *collection_type* with element type `E` is referred to here as *`col<E>`*.
+
+A *collection_type* exists at compile time only; a *collection_type* cannot be referenced in source or metadata.
+
+The [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) of a *collection_type* `col<E>` is `E`.
+
+An implicit *collection_type conversion* exists from a type with an [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) `Tₑ` to the *collection_type* `col<Tₑ>`.
```

Comment:
```suggestion
An implicit *collection type conversion* exists from a type with an [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) `Tₑ` to the *collection_type* `col<Tₑ>`.
```

Reply:
do you want to refer to iteration type?  or should we say `element type`? (just thinking upon the last few ldm talks).